### PR TITLE
[ONNX] Enable tensordot symbolic function.

### PR DESCRIPTION
**Pull request by fatcat-z**
Support tensordot in symbolic function of opset 12, and add tests accordingly.

### CODE REVIEW

**Review by jiafatom — APPROVED** [review]
LGTM, waiting for CI build.

File: torch/onnx/symbolic_opset12.py [hunk 1]
```diff
@@ -209,3 +211,58 @@ def unfold(g, input, dimension, size, step):
...
+
+    dim_count_a = sym_help._get_tensor_rank(input_a)
+    if dim_count_a is None:
+        raise RuntimeError('Unsupported: ONNX export of tensordot for tensor(input_a) of unknown rank.')
+
+    dim_count_b = sym_help._get_tensor_rank(input_b)
+    if dim_count_b is None:
+        raise RuntimeError('Unsupported: ONNX export of tensordot for tensor(input_b) of unknown rank.')
+
+    dims_a = [(dims_a[i] + dim_count_a)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset12.py:228]: nit: `dims_a = [dim + dim_count_a if dim < 0 else dim for dim in dims_a]`

File: torch/onnx/symbolic_opset12.py [hunk 2]
```diff
@@ -209,3 +211,58 @@ def unfold(g, input, dimension, size, step):
...
+              else dims_b[i]
+              for i in range(len(dims_b))]
+
+    left_dims_a = [i for i in range(dim_count_a) if (i not in dims_a)]
+    left_dims_b = [i for i in range(dim_count_b) if (i not in dims_b)]
+
+    new_input_a = permute(g, input_a, left_dims_a + dims_a)
+    new_input_b = permute(g, input_b, dims_b + left_dims_b)
+
+    intput_shape = g.op("Shape", new_input_a)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset12.py:244]:
```suggestion
    input_shape = g.op("Shape", new_input_a)
```

File: torch/onnx/symbolic_opset12.py [hunk 3]
```diff
@@ -209,3 +211,58 @@ def unfold(g, input, dimension, size, step):
...
+    left_sizes_a = sym_help._slice_helper(g, intput_shape, axes=[0], starts=[0], ends=[len(left_dims_a)])
+    shape_sizes = [left_sizes_a, g.op("Constant", value_t=torch.tensor([-1], dtype=torch.long))]
+    output_a = _reshape_from_tensor(g, new_input_a, shape_sizes)
+
+    intput_shape = g.op("Shape", output_a)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset12.py:249]:
```suggestion
    input_shape = g.op("Shape", output_a)
```

**Review by BowenBao — APPROVED** [review]
Looks good, minor comments.

File: torch/onnx/symbolic_opset12.py (hunk 2 above)
Comment (fatcat-z) [reply torch/onnx/symbolic_opset12.py]: Good catch.

File: torch/onnx/symbolic_opset12.py (hunk 1 above)
Comment (fatcat-z) [reply torch/onnx/symbolic_opset12.py]: Done.
